src/App.js

```diff
@@ -1,16 +1,33 @@
+import { useEffect, useState } from "react";
+import { fetchTopics, fetchArticles } from "./api";
 import "./App.css";
 import Articles from "./Components/Articles";
-import { useEffect, useState } from "react";
-import { fetchArticles } from "./api";
 import Header from "./Components/Header";
 import NavBar from "./Components/NavBar";
+import Topics from "./Components/Topics";
+import { Route, Routes } from "react-router-dom";
+import SingleTopic from "./Components/SingleTopic";
 
 function App() {
+  const [articles, setArticles] = useState([]);
+
   return (
     <div className="App">
       <Header />
       <NavBar />
-      <Articles />
+      <Routes>
+        <Route
+          path="/"
+          element={<Articles articles={articles} setArticles={setArticles} />}
+        ></Route>
+        <Route path="/topics" element={<Topics />}></Route>
+        <Route
+          path="/topics/:topic"
+          element={
+            <SingleTopic articles={articles} setArticles={setArticles} />
```

Comment:
It looks live you raised the articles up to App level, however they're not actually being used by the single article component as it is making its own API call. Have a look at lowering the articles state if you can :)

Reply:
I have changed this so that SingleTopic is not needed and everything routes through articles, please can you take a look and tell me if the way I've done it is what you meant?